pkg/steps/release/promote.go

```diff
@@ -71,107 +68,49 @@ func (s *promotionStep) Run(ctx context.Context, dry bool) error {
 		return fmt.Errorf("could not resolve pipeline imagestream: %v", err)
 	}
 
-	if len(s.config.Name) > 0 {
-		return retry.RetryOnConflict(promotionRetry, func() error {
-			is, err := s.dstClient.ImageStreams(s.config.Namespace).Get(s.config.Name, meta.GetOptions{})
-			if errors.IsNotFound(err) {
-				is, err = s.dstClient.ImageStreams(s.config.Namespace).Create(&imageapi.ImageStream{
-					ObjectMeta: meta.ObjectMeta{
-						Name:      s.config.Name,
-						Namespace: s.config.Namespace,
-					},
-				})
-			}
-			if err != nil {
-				return fmt.Errorf("could not retrieve target imagestream: %v", err)
-			}
-
-			for dst, src := range tags {
-				if valid, _ := findStatusTag(pipeline, src); valid != nil {
-					is.Spec.Tags = append(is.Spec.Tags, imageapi.TagReference{
-						Name: dst,
-						From: valid,
-					})
-				}
-			}
-
-			if dry {
-				istJSON, err := json.MarshalIndent(is, "", "  ")
-				if err != nil {
-					return fmt.Errorf("failed to marshal image stream: %v", err)
-				}
-				fmt.Printf("%s\n", istJSON)
-				return nil
-			}
-			if _, err := s.dstClient.ImageStreams(s.config.Namespace).Update(is); err != nil {
-				if errors.IsConflict(err) {
-					return err
-				}
-				return fmt.Errorf("could not promote image streams: %v", err)
-			}
-			return nil
-		})
+	if len(s.config.Name) == 0 {
+		return fmt.Errorf("name is a required field for release tagging")
```

Comment:
We should do this on config load

Reply:
I added it in the validation as well, I just wanted to make sure we had something here in case someone came in and got confused.